sql/core/src/test/scala/org/apache/spark/sql/streaming/TransformWithStateSuite.scala

```diff
@@ -1815,7 +1352,6 @@ class TransformWithStateSuite extends StateStoreMetricsTest
           TransformWithStateSuiteUtils.NUM_SHUFFLE_PARTITIONS.toString) {
         withTempDir { checkpointDir =>
           // When Avro is used, we want to set the StructFields to nullable
-          val shouldBeNullable = usingAvroEncoding()
```

Comment:
Looks like this is coupled with #49751. Shall we just do pure refactor here so that we can merge this first? Looks like #49751 needs some time to review.